src/admin/components/utilities/Config/index.tsx

```diff
@@ -3,10 +3,24 @@ import { SanitizedConfig } from '../../../../config/types';
 
 const Context = createContext<SanitizedConfig>({} as SanitizedConfig);
 
-export const ConfigProvider: React.FC<{config: SanitizedConfig, children: React.ReactNode}> = ({ children, config }) => (
-  <Context.Provider value={config}>
-    {children}
-  </Context.Provider>
-);
+export const ConfigProvider: React.FC<{config: SanitizedConfig, children: React.ReactNode}> = ({ children, config: incomingConfig }) => {
+  const [config, setConfig] = React.useState<SanitizedConfig>();
+
+  React.useEffect(() => {
+    const awaitConfig = async () => {
+      const resolvedConfig = await incomingConfig;
+      setConfig(resolvedConfig);
+    };
+    awaitConfig();
```

Comment:
This effect has a race condition: if the incoming config changes twice in quick succession, the following could happen:
1. awaitConfig() // 1
2. awaitConfig() // 2
3. setConfig() // 2
4. setConfig() // 1 (this one should be ignored)

Example: https://www.youtube.com/shorts/fXD9Ct03Q6k

Reply:
@thgh good call! This is done.